projects/knora-ui/src/lib/viewer/values/time-value/time-input/time-input.component.ts

```diff
@@ -0,0 +1,231 @@
+    this.stateChanges.next();
+  }
+
+  private _disabled = false;
+
+  @Input()
+  get placeholder() {
+    return this._placeholder;
+  }
+
+  set placeholder(plh) {
+    this._placeholder = plh;
+    this.stateChanges.next();
+  }
+
+  private _placeholder: string;
+
+  @Input() readonly = false;
+
+  @HostBinding('attr.aria-describedby') describedBy = '';
+
+  setDescribedByIds(ids: string[]) {
+    this.describedBy = ids.join(' ');
+  }
+
+  @Input()
+  get value(): string | null {
+    const userInput = this.form.value;
+    if (userInput.date !== null && userInput.time !== null) {
+      return this.userInputToTimestamp(userInput);
+    }
+    return null;
+  }
+
+  set value(timestamp: string | null) {
+    if (timestamp !== null) {
+      const dateTime = this.convertTimestampToDateTime(timestamp);
+      this.form.setValue({date: dateTime[0], time: dateTime[1]});
+    } else {
+      this.form.setValue({date: null, time: null});
+    }
+    this.stateChanges.next();
+  }
+
+  @Input() errorStateMatcher: ErrorStateMatcher;
+
+  constructor(fb: FormBuilder,
+              @Optional() @Self() public ngControl: NgControl,
+              private fm: FocusMonitor,
+              private elRef: ElementRef<HTMLElement>,
+              @Optional() _parentForm: NgForm,
+              @Optional() _parentFormGroup: FormGroupDirective,
+              _defaultErrorStateMatcher: ErrorStateMatcher) { 
+
+    super(_defaultErrorStateMatcher, _parentForm, _parentFormGroup, ngControl);
+
+    this.dateFormControl = new FormControl(null);
+    this.dateFormControl.setValidators([Validators.required]);
+    
+    this.timeFormControl = new FormControl({value: null, Validators: [Validators.required, this.timeValidator]});
+
+    this.form = fb.group({
+      date: this.dateFormControl,
+      time: this.timeFormControl
+    });
+
+    fm.monitor(elRef.nativeElement, true).subscribe(origin => {
+      this.focused = !!origin;
+      this.stateChanges.next();
+    });
+
+    if (this.ngControl != null) {
+      this.ngControl.valueAccessor = this;
+    }
+  }
+
+  ngDoCheck() {
+    if (this.ngControl) {
+      this.updateErrorState();
+    }
+  }
+
+  ngOnDestroy() {
+    this.stateChanges.complete();
+  }
+
+  onContainerClick(event: MouseEvent) {
+    if ((event.target as Element).tagName.toLowerCase() != 'input') {
+      this.elRef.nativeElement.querySelector('input').focus();
+    }
+  }
+
+  writeValue(datetime: string | null): void {
+    this.value = datetime;
+  }
+
+  registerOnChange(fn: any): void {
+    this.onChange = fn;
+  }
+
+  registerOnTouched(fn: any): void {
+    this.onTouched = fn;
+  }
+
+  setDisabledState(isDisabled: boolean): void {
+    this.disabled = isDisabled;
+  }
+
+  _handleInput(): void {
+    this.onChange(this.value);
+  }
+
+  // return converted Date obj as a string without the milliseconds
+  userInputToTimestamp(userInput: any): string {
+    let splitTime = userInput.time.split(":");
```

Comment:
When the user cancels the date, time seems to be an empty string:

```ts
{date: GregorianCalendarDate, time: ""}date: GregorianCalendarDate {calendarStart: CalendarDate, calendarEnd: CalendarDate, exactDate: true, jdnStart: 2458922, jdnEnd: 2458922, …}time: ""__proto__: Object

```

Reply:
fixed in bfec41b